guides/source/active_record_connection_management.md

```diff
@@ -0,0 +1,107 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON http://guides.rubyonrails.org.**
+
+Active Record Connection Management
+===================================
+
+In general, Active Record will automagically manage database connections in an
+effecient manner. See [Configuring a Database](/configuring.html#configuring-a-database) and [Database pooling](configuring.html#database-pooling) for information about the basics of optimally configuring your application to use available resources.
+
+Sometimes, you will need to manage Active Record connections and/or connecton
+pools yourself. These situations are:
+
+* Configuring a multi-process web server
+* Spawning processes
+
+In previous version of rails, you needed to think about managing connections when
+spawning threads, but this is no longer the case.
+
+--------------------------------------------------------------------------------
+
+Configuring a multi-process web server
+--------------------------------------
+
+Each process in a multi-process web server needs its own database connection pool.
+If the server is also multi-threaded, the threads within each process will automatically
+share that processes' connection pool.
+
+Here is how to configure a server to set up its pools while booting your app:
+
+### Before forking
+
+Before the server forks, you need to disconnect the pool so that the forked processes
+don't get confused. Here's how to do that in a puma config:
+
+```ruby
+before_fork do
+  ActiveRecord::Base.connection_pool.disconnect!
+end
+```
+
+### After forking
+
+After a process forks and has no connections, it needs to establish connections.
+Here's how to do that in a puma config:
+
+```ruby
+c.on_worker_boot do
+  ActiveRecord::Base.establish_connection
+end
+```
+
+Spawning processes
+------------------
+
+When spawning processes, you must do a small amount of manual database pool management.
+
+Let's say you you have a script that you are going to run as a working using `rails runner`,
+my_worker.rb. In this script, you are going to start two long-running processes.  Here's how you would go about doing that:
+
+```ruby
+ActiveRecord::Base.connection_pool.disconnect!
```

Comment:
Maybe you should mention that you won't be able to use the connection on the parent process anymore after this.
As you are disconnecting it.